packages/salesforcedx-vscode-apex-replay-debugger/src/breakpoints/checkpointService.ts

```diff
@@ -426,6 +422,166 @@ export class CheckpointService implements TreeDataProvider<BaseNode> {
+            });
+            const sourceLineInfoRetrieved: boolean = await retrieveLineBreakpointInfo();
+            // If we didn't get the source line information that'll be reported at that time, just return
+            if (!sourceLineInfoRetrieved) {
+              updateError = true;
+              return false;
+            }
+
+            // There can be a max of five active checkpoints
+            if (!checkpointService.hasFiveOrLessActiveCheckpoints(true)) {
+              updateError = true;
+              return false;
+            }
+
+            writeToDebuggerOutputWindow(
+              `${localizedProgressMessage}, ${nls.localize(
+                'checkpoint_creation_status_setting_typeref'
+              )}`
+            );
+            progress.report({
+              increment: 50,
+              message: localizedProgressMessage
+            });
+            // For the active checkpoints set the typeRefs using the source/line info
+            if (!setTypeRefsForEnabledCheckpoints()) {
+              updateError = true;
+              return false;
+            }
+
+            writeToDebuggerOutputWindow(
+              `${localizedProgressMessage}, ${nls.localize(
+                'checkpoint_creation_status_clearing_existing_checkpoints'
+              )}`
+            );
+            progress.report({
+              increment: 50,
+              message: localizedProgressMessage
+            });
+            // remove any existing checkpoints on the server
+            const allRemoved: boolean = await checkpointService.clearExistingCheckpoints();
+            if (!allRemoved) {
+              updateError = true;
+              return false;
+            }
+
+            writeToDebuggerOutputWindow(
+              `${localizedProgressMessage}, ${nls.localize(
+                'checkpoint_creation_status_uploading_checkpoints'
+              )}`
+            );
+            progress.report({
+              increment: 70,
+              message: localizedProgressMessage
+            });
+            // This should probably be batched but it makes dealing with errors kind of a pain
+            for (const cpNode of checkpointService.getChildren() as CheckpointNode[]) {
+              if (cpNode.isCheckpointEnabled()) {
+                if (
+                  !(await checkpointService.executeCreateApexExecutionOverlayActionCommand(
+                    cpNode
+                  ))
+                ) {
+                  updateError = true;
+                }
+              }
+            }
+
+            progress.report({
+              increment: 100,
+              message: localizedProgressMessage
+            });
+            writeToDebuggerOutputWindow(
+              `${localizedProgressMessage}, ${nls.localize(
+                'checkpoint_creation_status_processing_complete_success'
+              )}`
+            );
+          }
+        );
+      });
+    } finally {
+      writeToDebuggerOutputWindow(
```

Comment:
Even here, what's in the `finally` could be refactored into its own function.
Also, I think the syntax for finally is:
```
.finally(() => {
    // code
});
```

I haven't verified this, but that's the syntax I'm used to seeing.